src/object/srv_obj_migrate.c

```diff
@@ -2761,13 +2782,16 @@ migrate_start_ult(struct enum_unpack_arg *unpack_arg)
 			break;
 		d_list_del_init(&mrone->mo_list);
 
+		migrate_obj_get(arg);
+		mrone->mo_obj_arg = arg;
 		migrate_pool_tls_get(tls);
 		mrone->mo_tls = tls;
 
 		D_ASSERT(arg->tgt_idx == dss_get_module_info()->dmi_tgt_id);
 		rc = dss_ult_create(migrate_one_ult, mrone, DSS_XS_SELF, 0, MIGRATE_STACK_SIZE,
 				    NULL);
 		if (rc) {
+			migrate_obj_put(arg);
```

Comment:
how about just call migrate_obj_put(mrone->mo_obj_arg) within migrate_one_destroy? Because L3786 has already taken over the refcount